### PR TITLE
Fix DNI field - trigger required instead of visibility

### LINKED ISSUE #20321: Display DNI field in address BO, even if it's not required
#### Describe the bug

DNI (company ID) field is a regular part of address. 
In Prestashop, it only shows up in BO address edit, if you enable "DNI REQUIRED" checkbox in country settings.
In front office, it shows up normally.

#### Expected behavior

If dni field is set up in address format, it should be visible in BO regardless if its required or not. This nonsense is present ever since I am using prestashop... :-)

#### Steps to Reproduce

1. Set your country's address format to this:
_firstname lastname
company
dni
vat_number
address1
postcode city
Country:name
phone_
2. Go to FO, register a new account and see address form - DNI is there.
**3. Go to BO, edit the address you made and DNI is missing.**

#### The cause and fix
- After you open the address edit in BO or change the address country, js/admin.js makes an ajax call to AdminAddressesController asking, if DNI is required for this country. 
- If AdminAddressesController replies 1, js/admin.js hides the field. 
- It should instead add/remove required class from the input label and add/remove required="required" parameter from the input.

#### Additional information

* PrestaShop version: 1.7.6
* PHP version: 7.2

ping @matks Is it possible to merge a fix this to next minor version, or it needs to wait for a major and possibly a page migration? Shouldn't bring any BC breaks.

### PR DESCRIPTION
**Pull request by Hlavtox**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | There is some ancient code in address BO form, that hides DNI field, if it's not set as required in country settings. This fix controls the required state instead. In FO, its working correctly and showing up as optional.
| Type?         | bug fix / improvement
| Category?     | BO
| BC breaks?    | no
| Deprecations? |  no
| Fixed ticket? | Fixes #20321
| How to test?  |  see below

### How to test

1. Make two countries.
2. Set address format for both countries to this:
_firstname lastname
company
dni
vat_number
address1
postcode city
Country:name
phone_
3. Set one country to **DNI required** YES and the second one to NO.
4. Make two addresses, each with different country.
5. DNI field is required or not, as it should be, instead of hiding and showing.

### What I changed.
**form.tpl** - This is the initial state of the form. I changed it from setting display: none to the input to setting required, if needed.
**admin.js** - After loading the form or changing country, it asks adminaddresscontroller, if DNI is required for this country. If yes, it used to hide or show the input. I just changed the 2 lines to change the required class and property of the input.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/20325)

### CODE REVIEW

File: js/admin.js [hunk 1]
```diff
@@ -1409,9 +1409,11 @@ function dniRequired() {
     data: 'token=' + address_token + '&ajax=1&dni_required=1&tab=AdminAddresses&id_country=' + $('#id_country').val(),
     success: function(resp) {
       if (resp && resp.dni_required) {
-        $("#dni_required").fadeIn();
+        $("#dni_required label").addClass("required");
+        $("#dni_required input").prop('required', 'required');
```
Comment (Progi1984) [inline js/admin.js]:
```suggestion
        $('#dni_required label').addClass('required');
        $('#dni_required input').prop('required', 'required');
```

File: js/admin.js [hunk 2]
```diff
@@ -1409,9 +1409,11 @@ function dniRequired() {
...
       } else {
-        $("#dni_required").fadeOut();
+        $("#dni_required label").removeClass("required");
+        $("#dni_required input").removeProp('required');
```
Comment (Progi1984) [inline js/admin.js]:
```suggestion
        $('#dni_required label').removeClass('required');
        $('#dni_required input').removeProp('required');
```

**Review by Progi1984 — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]
thanks @Hlavtox !

**Review by PierreRambaud — APPROVED** [review]

**Comment by Hlavtox** [comment]
Hi guys, just realized this page was migrated and the code changes are useless. 😁

Unless targeting 1.7.6 version is unrealistic, we can close this.

**Comment by NeOMakinG** [comment]
Hey @Hlavtox,

Closing it then, thanks for your contribution anyway :)

**Pull request closed without merging**